README.md

```diff
@@ -83,7 +83,28 @@ val key = Secrets().getYourSecretKeyName(packageName)
 final String key = new Secrets().getYourSecretKeyName(getPackageName());
 ```
 
-# 4 - (Optional) Improve your key security
+# 4.1 - (Optional) Generate secrets from properties file
+You can generate secrets from properties file as well. 
+
+## Use case
+If you are using CI system to provide secrets, that are not hard-coded into the repository itself. It will re-generate those secrets in the app and build it. 
+
+This is useful if you want to split production keys from repository itself, thus increasing security in your project repository. 
+
+## Setting up
+1. Create properties file in root project directory (or CI creates one). For ex.: `creds.properties`
+2. Fill in wanted secrets. For ex.: 
+
+``` java-properties
+secret1=property123
+secret2=property321
+```
+
+3. Run `./gradlew hideSecretFromProps -PfromProps=creds.properties`
+It will regenerate all secret files in the project and update all secrets from the properties file.
+
+Make sure you are *not holding all the secrets in file which are related to project* (For ex.: Key / pass for signing the APK). Because it will generate all the keys and build into the app. 
```

Comment:
Instead of this explanation maybe it would be better to say `Create **a new** properties file in root project directory` on line 95 ?

Reply:
Hmm, this one is more like our use case. It is good to know, that the person who is adding these secrets, would not be adding *all* the project secrets to this file, only app related ones. But at the moment, I'm considering just removing this line entirely, as I believe this just adds a bit of confusion 👍

Reply:
Oh, and yeah, I've changed wording a bit by your suggestion as well 👍